### ISSUE TITLE
Update COp documentation to make it clearer

### ISSUE THREAD

**Issue by abergeron** [Theano/Theano#2361]
The changes introduced in #2356 to COp include some documentation changes, but it could still be much better and much clearer.

**Comment by nouiz** [comment]
Document get_op_params from the commit linked above.

**Comment by lamblin** [comment]
@abergeron Is this ticket still valid, or was that fixed since?

**Comment by abergeron** [comment]
I think it's still valid.

**Comment by lamblin** [comment]
@abergeron If this is still needed, could you have a look and check what needs to be done?

**Comment by abergeron** [comment]
I don't think we should document get_op_params since it is deprecated now that we have ParamsType.  What we should instead do is remove all instances of it and remove the supporting code.

As for the rest of the ticket, I'll have a look when I have time.
